core/src/main/java/org/web3j/ens/EnsResolver.java

```diff
@@ -78,40 +105,193 @@ protected PublicResolver obtainPublicResolver(String ensName) {
             } catch (Exception e) {
                 throw new EnsResolutionException("Unable to determine sync status of node", e);
             }
+        } else {
+            throw new EnsResolutionException("EnsName is invalid: " + ensName);
+        }
+    }
 
+    /**
+     * Provides an access to a valid offchain resolver in order to access other API methods.
+     *
+     * @param ensName our user input ENS name
+     * @return OffchainResolver
+     */
+    protected OffchainResolver obtainOffchainResolver(String ensName) {
+        if (isValidEnsName(ensName, addressLength)) {
+            try {
+                if (!isSynced()) {
+                    throw new EnsResolutionException("Node is not currently synced");
+                } else {
+                    return lookupOffchainResolver(ensName);
+                }
+            } catch (Exception e) {
+                throw new EnsResolutionException("Unable to determine sync status of node", e);
+            }
         } else {
             throw new EnsResolutionException("EnsName is invalid: " + ensName);
         }
     }
 
+    /**
+     * Returns the address of the resolver for the specified node.
+     *
+     * @param ensName The specified node.
+     * @return address of the resolver.
+     */
     public String resolve(String ensName) {
-
         if (Strings.isBlank(ensName) || (ensName.trim().length() == 1 && ensName.contains("."))) {
             return null;
         }
 
-        if (isValidEnsName(ensName, addressLength)) {
-            PublicResolver resolver = obtainPublicResolver(ensName);
+        try {
+            if (isValidEnsName(ensName, addressLength)) {
+                OffchainResolver resolver = obtainOffchainResolver(ensName);
 
-            byte[] nameHash = NameHash.nameHashAsBytes(ensName);
-            String contractAddress;
-            try {
-                contractAddress = resolver.addr(nameHash).send();
-            } catch (Exception e) {
-                throw new RuntimeException(
-                        "ENS resolver exception, unable to execute request: ", e);
-            }
+                boolean supportWildcard =
+                        resolver.supportsInterface(EnsUtils.ENSIP_10_INTERFACE_ID).send();
+
+                byte[] nameHash = NameHash.nameHashAsBytes(ensName);
+                String dnsEncoded = NameHash.dnsEncode(ensName);
+
+                String contractAddress;
+
+                if (supportWildcard) {
+                    String callData = resolver.addr(nameHash).encodeFunctionCall();
+                    String to = resolver.getContractAddress();
```

Comment:
assigned but no further used🤔  
Didn't you want to assigned this to contractAddress above?